### PR TITLE
AD-1(department module)

### PR DESCRIPTION
**Pull request by gshubha**
In this module for students as a user, we have created the UI that shows the working of different departments and various other functionalities like filing complaints about a particular department and other things. A pdf of required work and screenshots of UI are in the folder(\applications\department\UI and details)

### CODE REVIEW

File: FusionIIIT/applications/department/models.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from django.db import models
+
```
Comment (hvsblogger) [inline FusionIIIT/applications/department/models.py:4]: special request table is not present as mentioned in DB Schema

File: FusionIIIT/Fusion/settings/development.py [hunk 2]
```diff
@@ -6,13 +6,23 @@
...
+# }
+
 DATABASES = {
     'default': {
         'ENGINE': 'django.db.backends.postgresql_psycopg2',
         'NAME': 'fusionlab',
-        'HOST': '172.27.16.216',
-        'USER': 'fusion_admin',
-        'PASSWORD': 'hello123',
+        'HOST': 'localhost',
```
Comment (adi0311) [inline FusionIIIT/Fusion/settings/development.py:23]: Please don't commit or push this file. Remove it from the staging area before making any commit.

File: FusionIIIT/applications/central_mess/migrations/0018_merge_20210303_2016.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Generated by Django 3.1.5 on 2021-03-03 20:16
+
+from django.db import migrations
```
Comment (adi0311) [inline FusionIIIT/applications/central_mess/migrations/0018_merge_20210303_2016.py:3]: Don't push migrations. It is already told to not to do this. If you are using local DB, just simply remove these files from the staging area before commit.

File: FusionIIIT/applications/department/urls.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+urlpatterns = [
+
+    url(r'^$', views.hod, name='dep'),
+    url(r'^file_request/$', views.file_request, name='file_request'),
+    url(r'^BtechFirstYear_Students/$', views.BtechFirstYear_Students, name='BtechFirstYear_Students'),
+    url(r'^BtechSecondYear_Students/$', views.BtechSecondYear_Students, name='BtechSecondYear_Students'),
+    url(r'^BtechThirdYear_Students/$', views.BtechThirdYear_Students, name='BtechThirdYear_Students'),
+    url(r'^BtechFinalYear_Students/$', views.BtechFinalYear_Students, name='BtechFinalYear_Students'),
+<<<<<<< HEAD
```
Comment (adi0311) [inline FusionIIIT/applications/department/urls.py]: Don't push the code that can break the application. Resolve the conflicts properly. VS Code has an excellent feature to resolve conflict. Make sure you do this before committing your code.

File: FusionIIIT/applications/department/views.py [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+#from .models import ()
+from jsonschema import validate
+from jsonschema.exceptions import ValidationError
+
+# Create your views here.
+
+def index(request):
+    context = {}
+    print(request.user)
```
Comment (adi0311) [inline FusionIIIT/applications/department/views.py]: Remember final code should not have any print statements or unnecessary comments. Use loggers(that too when there might be some errors, for example with try-except).

File: FusionIIIT/applications/department/views.py [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+    if user_designation == "student":
+        return render(request,"department/index.html")
+    elif(str(user.extrainfo.user_type)=='faculty'):
+        return render(request, 'department/dep_request.html', {"user_designation":'faculty'})
+
+def file_request(request):
+    #return render(request, "department/dep_complaint.html")
+    return render(request, 'department/dep_request.html')
+
+def BtechFirstYear_Students(request):
```
Comment (adi0311) [inline FusionIIIT/applications/department/views.py]: Please, it's been told several times to follow PEP-8. If you couldn't find that doc, here is the [link](https://www.python.org/dev/peps/pep-0008/)
Also, to summarize for you, take care of these points:-

- Make sure each view function and model class should have a docstring.
- Class name is in PascalCase.
- Functions, variables name should be in snake_case.

**Review by adi0311 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**